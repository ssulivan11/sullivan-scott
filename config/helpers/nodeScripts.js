/* eslint-disable no-console, @typescript-eslint/no-var-requires */
const branch = require('git-branch')
const shell = require('shelljs')
const fs = require('fs')
const path = require('path')
require('colors')

module.exports.statusCheck = () => {
  shell.exec('git status -s', { silent: true }, (code, stdout) => {
    const monitoredFiles = ['.snap', '.bundlesizeconfig']
    const noInvalidUncommited = !monitoredFiles.some((elem) => stdout.includes(elem))
    if (stdout.length === 0 || noInvalidUncommited) {
      console.log(`${`\n ✔`.green}  Git Status Check Valid`)
      shell.exec(process.exit(0))
    } else {
      console.log(
        '\n  ERROR: Looks like you have an uncommited snapshot change or bundlesize update,\n         please commit this change before pushing.\n'
          .red,
      )
      process.exit(1)
    }
  })
}

module.exports.branchCheck = () => {
  branch((err, name) => {
    if (err) throw err
    const validBranchDirs = ['feature/', 'refactor/', 'bug/']
    let validName = false
    for (let i = 0; i !== validBranchDirs.length; i += 1) {
      if (name.indexOf(validBranchDirs[i]) !== -1) validName = true
    }
    if (!validName) {
      console.log(`${`\n ❌  Git Branch Name Error:`} - ${`${name}`.grey} \n`)
      console.log(`    ▶ Your branch name does not match the valid checks, please use of the following:`)
      console.log(`      ${validBranchDirs}\n`.blue)
      shell.exec(process.exit(1))
    } else {
      console.log(`${`\n ✔`.green}  Git Branch Name Valid - ${`${name}`.grey} \n`)
      shell.exec(process.exit(0))
    }
  })
}

module.exports.moveAssets = () => {
  const moveFiles = (files) => {
    for (let i = 0; i < files.length; i += 1) {
      const file = files[i]
      const f = path.basename(file)
      const source = fs.createReadStream(file)
      const dest = fs.createWriteStream(path.resolve('./build/', f))

      source.pipe(dest)
      source.on('end', () => {
        console.log(`${`✔`.green}  Copied - ${files[i]}`)
      })
      source.on('error', (err) => {
        console.log(err)
      })
    }
  }
  const filesToMove = ['./src/assets/manifest.json', './src/assets/favicon.ico', './src/assets/service-worker.js']
  moveFiles(filesToMove)
}
