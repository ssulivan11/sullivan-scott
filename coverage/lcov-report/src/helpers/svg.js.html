<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/helpers/svg.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/helpers</a> svg.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">const github =
  'M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12'
const linkedIn =
  'M20.447 20.452h-3.554v-5.569c0-1.328-.027-3.037-1.852-3.037-1.853 0-2.136 1.445-2.136 2.939v5.667H9.351V9h3.414v1.561h.046c.477-.9 1.637-1.85 3.37-1.85 3.601 0 4.267 2.37 4.267 5.455v6.286zM5.337 7.433c-1.144 0-2.063-.926-2.063-2.065 0-1.138.92-2.063 2.063-2.063 1.14 0 2.064.925 2.064 2.063 0 1.139-.925 2.065-2.064 2.065zm1.782 13.019H3.555V9h3.564v11.452zM22.225 0H1.771C.792 0 0 .774 0 1.729v20.542C0 23.227.792 24 1.771 24h20.451C23.2 24 24 23.227 24 22.271V1.729C24 .774 23.2 0 22.222 0h.003z'
const codePen =
  'M24 8.182l-.018-.087-.017-.05c-.01-.024-.018-.05-.03-.075-.003-.018-.015-.034-.02-.05l-.035-.067-.03-.05-.044-.06-.046-.045-.06-.045-.046-.03-.06-.044-.044-.04-.015-.02L12.58.19c-.347-.232-.796-.232-1.142 0L.453 7.502l-.015.015-.044.035-.06.05-.038.04-.05.056-.037.045-.05.06c-.02.017-.03.03-.03.046l-.05.06-.02.06c-.02.01-.02.04-.03.07l-.01.05C0 8.12 0 8.15 0 8.18v7.497c0 .044.003.09.01.135l.01.046c.005.03.01.06.02.086l.015.05c.01.027.016.053.027.075l.022.05c0 .01.015.04.03.06l.03.04c.015.01.03.04.045.06l.03.04.04.04c.01.013.01.03.03.03l.06.042.04.03.01.014 10.97 7.33c.164.12.375.163.57.163s.39-.06.57-.18l10.99-7.28.014-.01.046-.037.06-.043.048-.036.052-.058.033-.045.04-.06.03-.05.03-.07.016-.052.03-.077.015-.045.03-.08v-7.5c0-.05 0-.095-.016-.14l-.014-.045.044.003zm-11.99 6.28l-3.65-2.44 3.65-2.442 3.65 2.44-3.65 2.44zm-1.034-6.674l-4.473 2.99L2.89 8.362l8.086-5.39V7.79zm-6.33 4.233l-2.582 1.73V10.3l2.582 1.726zm1.857 1.25l4.473 2.99v4.82L2.89 15.69l3.618-2.417v-.004zm6.537 2.99l4.474-2.98 3.613 2.42-8.087 5.39v-4.82zm6.33-4.23l2.583-1.72v3.456l-2.583-1.73zm-1.855-1.24L13.042 7.8V2.97l8.085 5.39-3.612 2.415v.003z'
&nbsp;
export { github, linkedIn, codePen }
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Aug 19 2019 11:51:04 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
