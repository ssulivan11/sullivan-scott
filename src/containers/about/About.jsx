import React from 'react'
import PropTypes from 'prop-types'
// import { motion, AnimatePresence } from 'framer-motion'
import style from './about.scss'

const About = ({ windowSize, active, 'data-test': dataTest }) => {
  return (
    <div style={{ minHeight: windowSize.height, position: 'relative' }} data-test={dataTest}>
      <div
        className={`container ${style.about} ${active === 'about' ? style['about-active'] : style['about-inactive']}`}>
        <img
          positionTransition
          key='bio-image'
          className={style['bio-pic']}
          width='150'
          height='150'
          initial={{ scale: 1 }}
          animate={{ scale: 2 }}
          alt="Scott's profile pic"
          src='https://scontent-lhr3-1.cdninstagram.com/vp/c5cdb50d8b3a1286724a1f27f8c125ab/5DA5F673/t51.2885-15/e35/14134966_1175689132493497_1840210486_n.jpg?_nc_ht=scontent-lhr3-1.cdninstagram.com'
        />
        <h2 className={style['bio-heading']}>Hello, I’m Scott, nice to meet you.</h2>
        <p className={style['bio-text']}>
          I&#39;m a Sr UI Engineer, specializing in Javascript, Node.js, front end libraries and frameworks.
        </p>

        <div className={style['bio-grid']}>
          <div className={style['bio-grid-item']}>
            <h3 className={style['bio-grid-title']}>1. Developing</h3>
            <ul>
              <li>React · Redux</li>
              <li>CSS (scss, styled-components, stylus)</li>
              <li>Web Performance, Webpack & Bundlesize management</li>
            </ul>
          </div>
          <div className={style['bio-grid-item']}>
            <h3 className={style['bio-grid-title']}>2. Architecture</h3>
            <ul>
              <li>Global JS libraries & styleguide patterns</li>
              <li>CMS Headless integration</li>
              <li>Git standards and Peer Review</li>
            </ul>
          </div>
          <div className={style['bio-grid-item']}>
            <h3 className={style['bio-grid-title']}>3. Engineering</h3>
            <ul>
              <li>Node.js, Express.js, and SSR</li>
              <li>Jenkins & CI development</li>
              <li>Docker Container management</li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  )
}

About.propTypes = {
  windowSize: PropTypes.shape({
    height: PropTypes.number,
    width: PropTypes.number,
  }),
  active: PropTypes.string,
  'data-test': PropTypes.string,
}

export default About
