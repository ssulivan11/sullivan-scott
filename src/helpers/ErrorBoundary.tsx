import * as React from 'react'

interface Props {
  callbackFunction: object
  retries: number
  children: JSX.Element[] | JSX.Element
}

interface State {
  hasError: boolean
  errorCount: number
}

class ErrorBoundary extends React.Component<Props, State> {
  static defaultProps = {
    retries: 0,
    callbackFunction: () => {},
  }

  constructor(props) {
    super(props)
    this.state = {
      hasError: false,
      errorCount: 0,
    }
  }

  componentDidCatch(error, errorInfo) {
    const { retries, callbackFunction } = this.props
    if (retries >= 1) {
      this.setState((oldState) => ({ errorCount: oldState.errorCount + 1 }))
    } else {
      this.setState({
        hasError: true,
      })
      // Object.keys(errorInfo).forEach((key) => {
      //   console.warn(key, errorInfo[key])
      // })
      if (typeof callbackFunction === 'function') callbackFunction()
    }
  }

  render() {
    const { children, retries } = this.props
    const { errorCount, hasError } = this.state

    if ((retries >= 1 && errorCount > retries) || hasError) return null
    return children
  }
}

export default ErrorBoundary
