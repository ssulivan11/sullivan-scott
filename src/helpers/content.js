const content = {
  title: 'Scott Sullivan UI Engineer',
  description: 'Scott Sullivan UI Engineer',
  canonical: 'https://ssullivan.info/',
  mainHeadingText: 'Scott Sullivan',
  subHeadingText: 'UI Engineer - London / Chicago',
  bioHeading: 'Hello and welcome',
  bioText:
    "I'm a Senior Software Engineer who is meticulous about quality, meeting requirements within timelines and pushing technical boundaries to deliver outstanding user experiences.",
  bioImg: 'images/profile-pic.jpg',
  bioFavorites: '',
  skills: [
    {
      title: 'Development',
      items: [
        'React | Redux',
        'CSS, SCSS, Styled-Components',
        'Web Performance | PWA',
        'Webpack & Bundle-size Management',
        'Jest Unit Tests w/ Enyme',
        'Cypress E2E & Nightwatch',
      ],
    },
    {
      title: 'Architecture',
      items: [
        'Boilerplate & Component Scaffolding',
        'Typescript',
        'Component based Storybooks',
        'Headless CMS Integration',
        'Style-guide Patterns',
        'Git | Gitflow | Peer Review',
      ],
    },
    {
      title: 'Infrastructure',
      items: [
        'Node.js, Express.js & Hapi.js',
        'SSR Applications',
        'Jenkins CI',
        'Docker Containerization ',
        'Global Package Management',
        'Monorepo & SemVer Practice',
      ],
    },
  ],
  social: {
    linkedIn: 'https://www.linkedin.com/in/scott-j-sullivan',
    github: 'https://github.com/ssulivan11',
    codePen: 'https://codepen.io/ssulivan11',
  },
}

module.exports = content
