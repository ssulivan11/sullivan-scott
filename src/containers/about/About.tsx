import React from 'react'
import { motion, useViewportScroll, useTransform } from 'framer-motion'

import * as style from './about.scss'

interface AboutProps {
  windowSize: {
    height: number
    width: number
  }
  skills: [{ items: [string]; title: string }]
  bioText: string
  bioHeading: string
  active?: string
  bioImg: string
  'data-test': string
}

const About: React.FunctionComponent<AboutProps> = ({
  windowSize,
  active,
  skills,
  bioHeading,
  bioText,
  bioImg,
  'data-test': dataTest,
}) => {
  const { scrollYProgress } = useViewportScroll()
  const circle1Scale = useTransform(scrollYProgress, [0, 1], [0.2, 2])
  const circle2Scale = useTransform(scrollYProgress, [0, 1], [0.1, 2])
  return (
    <div className={style['about-wrapper']} style={{ minHeight: windowSize.height }} data-test={dataTest}>
      <main
        role='main'
        className={`container ${style.about} ${active === 'about' ? style['about-active'] : style['about-inactive']}`}>
        <div className={style['bio-pic']}>
          <motion.span
            style={{
              scale: circle1Scale,
            }}
          />
          <motion.span
            style={{
              scale: circle2Scale,
            }}
          />
          <img key='bio-pic' width='150' height='150' alt="Scott's profile pic" src={bioImg} />
        </div>

        <h2 className={style['bio-heading']} data-test='bio-heading'>
          {bioHeading}
        </h2>
        <p className={style['bio-text']} data-test='bio-text'>
          {bioText}
        </p>
        <div className={style['bio-grid']}>
          {skills.map((skill) => {
            return (
              <div className={style['bio-grid-item']} key={skill.title}>
                <h3 className={style['bio-grid-title']}>{skill.title}</h3>
                <ul tabIndex={0}>
                  {skill.items.map((item) => (
                    <li key={item}>{item}</li>
                  ))}
                </ul>
              </div>
            )
          })}
        </div>
      </main>
    </div>
  )
}

export default About
