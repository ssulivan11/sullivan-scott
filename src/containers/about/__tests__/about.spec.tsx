import * as React from 'react'
import { shallow, mount } from 'enzyme'

import About from '../About'
import * as content from '../../../helpers/content'

const defaultTestProps = {
  skills: content.skills,
  bioText: content.bioText,
  bioHeading: content.bioHeading,
  bioImg: content.bioImg,
  'data-test': 'contact',
  active: 'about',
  windowSize: {
    height: 500,
    width: 0,
  },
}

const inactiveProps = {
  ...defaultTestProps,
  active: 'home',
}

const setup = (renderer = mount, props) => {
  const component = renderer(<About {...props} />)
  return {
    getInstance: () => component,
    getAbout: () => component.find(About),
    debug: () => component.debug(),
  }
}

let wrapper
describe('<About />', () => {
  it('active container', () => {
    wrapper = setup(shallow, defaultTestProps)
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('inactive container', () => {
    wrapper = setup(shallow, inactiveProps)
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
})
