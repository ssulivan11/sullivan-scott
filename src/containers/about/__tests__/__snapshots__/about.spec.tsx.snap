// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> active container 1`] = `
Object {
  "debug": [Function],
  "getAbout": [Function],
  "getInstance": [Function],
}
`;

exports[`<About /> inactive container 1`] = `
Object {
  "debug": [Function],
  "getAbout": [Function],
  "getInstance": [Function],
}
`;
